CRM/Batch/BAO/EntityBatch.php

```diff
@@ -29,19 +29,14 @@ public static function create($params) {
   /**
    * Remove entries from entity batch.
    * @param array|int $params
+   * @deprecated
    * @return CRM_Batch_DAO_EntityBatch
    */
   public static function del($params) {
     if (!is_array($params)) {
       $params = ['id' => $params];
     }
-    $entityBatch = new CRM_Batch_DAO_EntityBatch();
-    $entityId = $params['id'] ?? NULL;
```

Comment:
As alarming as this is - I wonder if there is a reason for it

Reply:
I suspect the "reason" is careless use of `CRM_Utils_Array::value()` which historically was used more than necessary, and recently would have been auto-converted to `??`.

Reply:
Oh fudge I was wrong:
https://github.com/civicrm/civicrm-core/blob/5.40/CRM/Financial/Page/AJAX.php#L473-L483
Ok I've updated the code to lookup the id when needed.

Reply:
@colemanw I think that is still wrong - I think that is a multiple row delete - perhaps replace that with a v4 api call & standardise delete? (We should do a universe search but could keep it 'liight' since people know they need to use the api for crud functions)

Reply:
Crap. Ok I'll revert that file from this PR & we can tackle it separate.

Reply:
good plan

Reply:
@eileenmcnaughton see #21213